conn_handshake.go

```diff
@@ -25,39 +25,32 @@ import (
 )
 
 func (c *connect) handshake(database, username, password string) error {
+	defer c.buffer.Reset()
 	c.debugf("[handshake] -> %s", proto.ClientHandshake{})
 	c.conn.SetDeadline(time.Now().Add(c.opt.DialTimeout))
 	defer c.conn.SetDeadline(time.Time{})
 	{
-		c.encoder.Byte(proto.ClientHello)
-		if err := (&proto.ClientHandshake{}).Encode(c.encoder); err != nil {
-			return err
-		}
+		c.buffer.PutByte(proto.ClientHello)
```

Comment:
you can use chproto.ClientHello here

Reply:
Better to just rip out handshake code completely on next PR